File: Version C#/Ejercicio2.cs
using System;

// Clase con encapsulación para proteger los datos
public class Empleado
{
    // Atributos privados: no pueden ser modificados directamente
    // desde fuera de la clase.
    private string _nombre;
    private int _edad;

    // Propiedad Nombre
    public string Nombre
    {
        get { return _nombre; }
        set { _nombre = value; }
    }

    // Propiedad Edad con validación
    public int Edad
    {
        get { return _edad; }
        set
        {
            // La edad debe ser mayor que 0 y menor que 100
            if (value > 0 && value < 100)
            {
                _edad = value;
            }
            else
            {
                throw new ArgumentException(
                    "La edad debe ser mayor que 0 y menor que 100."
                );
            }
        }
    }
}

public class Ejercicio2
{
    public static void Main(string[] args)
    {
        Console.Clear();

        Console.WriteLine("========================================");
        Console.WriteLine(" EJERCICIO 2: ENCAPSULACIÓN Y CONTROL");
        Console.WriteLine("             DE ACCESO");
        Console.WriteLine("========================================");

        // Crear un objeto de tipo Empleado
        Empleado emp = new Empleado();

        // Solicitar el nombre al usuario
        Console.Write("\nIngrese el nombre del empleado: ");
        emp.Nombre = Console.ReadLine();

        // Solicitar la edad al usuario
        while (true)
        {
            Console.Write("Ingrese la edad del empleado: ");

            // Primero comprobamos que el usuario haya ingresado
            // un número entero válido.
            if (!int.TryParse(Console.ReadLine(), out int edad))
            {
                Console.WriteLine(
                    "[Error] Debe ingresar un número entero."
                );

                continue;
            }

            try
            {
                // La propiedad Edad recibe el valor y se encarga
                // de determinar si es válido.
                emp.Edad = edad;

                // Si no se produjo ningún error, la edad es válida.
                break;
            }
            catch (ArgumentException ex)
            {
                // Mostrar el mensaje generado por la propiedad
                // cuando la edad no cumple las condiciones.
                Console.WriteLine($"[Error] {ex.Message}");
            }
        }

        // Mostrar los datos del empleado
        Console.WriteLine("\n========================================");
        Console.WriteLine("       EMPLEADO REGISTRADO");
        Console.WriteLine("========================================");

        Console.WriteLine($"Nombre: {emp.Nombre}");
        Console.WriteLine($"Edad:   {emp.Edad}");

        Console.WriteLine("\nPresione ENTER para finalizar...");
        Console.ReadLine();
    }
}

File: Version Python/Ejercicio2.py
import tkinter as tk
from tkinter import messagebox


# ==========================================
# CLASE EMPLEADO
# ==========================================

class Empleado:
    def __init__(self):
        # Atributos privados
        self._nombre = ""
        self._edad = 0

    # ======================================
    # PROPIEDAD NOMBRE
    # ======================================

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, valor):
        self._nombre = valor

    # ======================================
    # PROPIEDAD EDAD CON VALIDACIÓN
    # ======================================

    @property
    def edad(self):
        return self._edad

    @edad.setter
    def edad(self, valor):
        # La validación pertenece a la clase,
        # no a la interfaz gráfica.
        if valor > 0 and valor < 100:
            self._edad = valor
        else:
            raise ValueError(
                "La edad debe ser mayor que 0 y menor que 100."
            )


# ==========================================
# INTERFAZ GRÁFICA
# ==========================================

class Aplicacion:
    def __init__(self, ventana):

        self.ventana = ventana

        self.ventana.title(
            "Ejercicio 2 - Encapsulación"
        )

        self.ventana.geometry("500x500")
        self.ventana.resizable(False, False)

        # Crear objeto Empleado
        self.empleado = Empleado()

        # ==================================
        # TÍTULO
        # ==================================

        titulo = tk.Label(
            ventana,
            text="EJERCICIO 2: ENCAPSULACIÓN",
            font=("Arial", 18, "bold")
        )

        titulo.pack(pady=20)

        subtitulo = tk.Label(
            ventana,
            text="Registro de empleado",
            font=("Arial", 12)
        )

        subtitulo.pack()

        tk.Label(
            ventana,
            text="-" * 50
        ).pack(pady=10)

        # ==================================
        # NOMBRE
        # ==================================

        tk.Label(
            ventana,
            text="Nombre del empleado:"
        ).pack(pady=5)

        self.entrada_nombre = tk.Entry(
            ventana,
            width=35
        )

        self.entrada_nombre.pack()

        # ==================================
        # EDAD
        # ==================================

        tk.Label(
            ventana,
            text="Edad del empleado:"
        ).pack(pady=10)

        self.entrada_edad = tk.Entry(
            ventana,
            width=35
        )

        self.entrada_edad.pack()

        # ==================================
        # BOTÓN REGISTRAR
        # ==================================

        tk.Button(
            ventana,
            text="Registrar empleado",
            width=25,
            command=self.registrar_empleado
        ).pack(pady=20)

        # ==================================
        # RESULTADO
        # ==================================

        self.resultado = tk.Label(
            ventana,
            text="Ingrese los datos del empleado.",
            font=("Arial", 12),
            justify="center"
        )

        self.resultado.pack(pady=20)

        # ==================================
        # BOTÓN LIMPIAR
        # ==================================

        tk.Button(
            ventana,
            text="Limpiar",
            width=25,
            command=self.limpiar
        ).pack(pady=5)

        # ==================================
        # BOTÓN SALIR
        # ==================================

        tk.Button(
            ventana,
            text="Salir",
            width=25,
            command=ventana.destroy
        ).pack(pady=5)

    # ======================================
    # REGISTRAR EMPLEADO
    # ======================================

    def registrar_empleado(self):

        nombre = self.entrada_nombre.get().strip()

        # Validar que se haya ingresado un nombre
        if not nombre:
            messagebox.showerror(
                "Error",
                "Debe ingresar el nombre del empleado."
            )

            return

        # Obtener la edad
        texto_edad = self.entrada_edad.get().strip()

        # Verificar que sea un número entero
        try:
            edad = int(texto_edad)

        except ValueError:
            messagebox.showerror(
                "Error",
                "La edad debe ser un número entero."
            )

            return

        # ==================================
        # ASIGNAR DATOS MEDIANTE PROPIEDADES
        # ==================================

        self.empleado.nombre = nombre

        try:
            # La propiedad edad se encarga de validar
            # el valor recibido.
            self.empleado.edad = edad

        except ValueError as error:
            messagebox.showerror(
                "Edad no válida",
                str(error)
            )

            return

        # ==================================
        # MOSTRAR RESULTADO
        # ==================================

        self.resultado.config(
            text=(
                "Empleado registrado correctamente.\n\n"
                f"Nombre: {self.empleado.nombre}\n"
                f"Edad: {self.empleado.edad}"
            )
        )

        messagebox.showinfo(
            "Registro exitoso",
            "El empleado ha sido registrado correctamente."
        )

    # ======================================
    # LIMPIAR CAMPOS
    # ======================================

    def limpiar(self):

        self.entrada_nombre.delete(0, tk.END)
        self.entrada_edad.delete(0, tk.END)

        self.resultado.config(
            text="Ingrese los datos del empleado."
        )


# ==========================================
# INICIAR APLICACIÓN
# ==========================================

ventana = tk.Tk()

app = Aplicacion(ventana)

ventana.mainloop()